docs/DeveloperGuide.md

```diff
@@ -239,6 +240,47 @@ _{more aspects and alternatives to be added}_
 _{Explain here how the data archiving feature will be implemented}_
 
 
+### Listing Events
+
+The List Events feature allows users to enter `listEvents` and update the UI display for events, showing a list of all 
+events sorted in their current order inside `ModelManager`. In the future, the command will take in a parameter,
+which specifies the field that the events list can be permanently sorted by.
+
+### Implementation
+The List Events feature is facilitated by `ListEventsCommand` which extends from Command. Additionally, it implements
+the following operation:
+
+* `ModelManager#updateFilteredEventList`  — Updates the predicate inside `ModelManager`'s filtered event list 
+to modify and sort which and how events are shown.
+
+This operation is exposed in the `Model` interface as the method `Model#updateFilteredEventList`.
+
+The following sequence diagram shows how the listEvents operation works.
+
+![ListEventsSequenceDiagram](images/ListEventsSequenceDiagram.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** Only the activities related to events are
+considered and shown in this activity diagram.
+</div>
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for ListEventsCommand should
+end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of the diagram.
+</div>
+
+### Design Considerations:
```

Comment:
I think this should be a 4th level heading as how it is for the undo/redo feature

```suggestion
#### Design Considerations:
```